src/main/java/org/sonarsource/sonarlint/ls/folders/WorkspaceFoldersManager.java

```diff
@@ -57,25 +74,30 @@ public void initialize(@Nullable List<WorkspaceFolder> workspaceFolders) {
 
   public void didChangeWorkspaceFolders(WorkspaceFoldersChangeEvent event) {
     LOG.debug("Processing didChangeWorkspaceFolders event");
+    var removedFolderWrappers = new ArrayList<WorkspaceFolderWrapper>();
+    var addedFolderWrappers = new ArrayList<WorkspaceFolderWrapper>();
     for (var removed : event.getRemoved()) {
       var uri = create(removed.getUri());
-      removeFolder(uri);
+      removedFolderWrappers.add(removeFolder(uri));
```

Comment:
Possibly adding `null` to the list, not sure this is good

Reply:
:+1: